### ISSUE TITLE
Order receptions: ErrorWarehouseRequiredIntoReceptionLine

### ISSUE THREAD

**Issue by Mavyre** [Dolibarr/dolibarr#13024]
# Bug
When receiving an order, we're encountering this error: "ErrorWarehouseRequiredIntoReceptionLine". It wasn't present in v10;  seems to be a regression.

## Environment
- **Version**: 11
- **OS**: Debian 9
- **Web server**: Apache2
- **PHP**: 7.3
- **Database**: MariaDB
- **URL(s)**: /reception/card.php?originid=xx&origin=supplierorder

## Expected and actual behavior
Should work as the Warehouse is specified. Currently, triggers the "ErrorWarehouseRequiredIntoReceptionLine" error

## Steps to reproduce the behavior
Create any order and try to receive products from it.

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
[*Files*]

![Capture d’écran 2020-02-04 à 14 30 50](https://user-images.githubusercontent.com/4330449/73751080-a0e3c100-475e-11ea-916e-ddcb09365b3f.png)

**Comment by eldy** [comment]
Can you confirm if you have enabled or not the module "Reception" ?
Can you provide a screenshot of full page you have just before clicking on button that show the message ?

**Comment by Mavyre** [comment]
Hi @eldy, I confirm that we have the Reception module. We enabled the Hidden Option that doesn't require a warehouse for a reception line and we noticed a strange behaviour: after a modification of warehouse, we have to refresh once before seeing the correct Warehouse.

I'll try to dig into it and give you a full bug report with everything I found. Using the hidden option as a workaround works for our folks for now.
